contentcuration/contentcuration/static/js/edit_channel/utils/number_parser.js

```diff
@@ -0,0 +1,111 @@
+/****************
+
+--- NUMBER REGULAR EXPRESSIONS ---
+VALID_NUMBER: [MIXED_NUMBER | FRACTION | DECIMAL | PERCENTAGE | INTEGER]
```

Comment:
Would be nice if you hadn't had to implement them in both languages, but I fear working out how to do that would be more trouble than it's worth.

Perhaps make a comment here referencing the Python parser too, so that future developers (including yourself) are reminded to update both?